### PR TITLE
Add io.gitlab.android_translation_layer.BaseApp

### PR DESCRIPTION
**Pull request by julianwi**
Hi, I'm involved in the android translation layer project (https://gitlab.com/android_translation_layer). We are reimplementing the Android application API on top of GNU/Linux and Freedesktop technology. NewPipe is currently one of the best supported apps (nearly feature complete). Therefore we would like to add it to Flathub. This is just the baseapp which can be used to package NewPipe and other Android applications in the future.

### Please confirm your submission meets all the criteria

- [x] Please describe the application briefly.
  
This new baseapp helps to package Android applications as Flatpak.

It contains the Android runtime (ART), core java libraries (libcore) the skia graphics library and a lightweight translation layer making Android applications compatible with XDG specs.

The size of the baseapp is about 48MB. The largest parts are libart, ICU4J, libskia and framework-res.apk, each of them about 10MB plus some smaller libraries. These libraries will be required by any Android application and are a perfect fit for a baseapp.
The first launch of an application using this baseapp will take a few seconds to do the ahead-of-time compilation of Java code. Subsequent launches are nearly instant.
With the help of android translation layer, Android applications integrate nicely into the Linux Desktop environment. Notifications are published to the XDG-desktop-portal, media playback can be controlled using MPRIS, the Light/Dark theme is automatically selected on startup, and the widget tree can be inspected using GTK-Inspector.
An example of how to use this baseapp can be found here: https://gitlab.com/julianwi/newpipe_flatpak/-/blob/master/newpipe/org.schabi.NewPipe.yml

- [x] The domain used for the application ID is [controlled by the application developers][appid-domain] and the [application id guidelines][appid] are followed.
- [x] I have read the and followed all the [Submission and licence requirements][reqs].
- [x] I have [built][build] and tested the submission locally.
- [x] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about this submission. **Link:** https://github.com/TeamNewPipe/NewPipe/issues/11597

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

File: io.gitlab.android_translation_layer.BaseApp.yml [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+        url: https://github.com/wolfSSL/wolfssl.git
+        tag: v5.7.0-stable
+  - name: vixl
+    buildsystem: meson
+    build-options:
+      config-opts:
+        - -Dsimulator=none
+    only-arches:
+      - aarch64
+      - arm
```
Comment (hfiguiere) [inline io.gitlab.android_translation_layer.BaseApp.yml:218]:
```suggestion
```

File: io.gitlab.android_translation_layer.BaseApp.yml [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+    sources:
+      - type: git
+        url: https://github.com/KhronosGroup/OpenXR-SDK.git
+        tag: 'release-1.1.36'
+  - name: skia
+    buildsystem: simple
+    build-commands:
+      - gn gen 'out/linux/x64' --args='is_official_build=true skia_enable_tools=false target_os="linux" target_cpu="x64" skia_use_icu=false skia_use_sfntly=false skia_use_piex=true skia_use_system_harfbuzz=true skia_use_system_expat=true skia_use_system_freetype2=true skia_use_system_libjpeg_turbo=true skia_use_system_libpng=true skia_use_system_libwebp=true skia_use_system_zlib=true skia_enable_gpu=true extra_cflags=[ "-DSKIA_C_DLL" ] linux_soname_version="99.9"'
+      - ninja -C 'out/linux/x64' SkiaSharp
+      - install -D out/linux/x64/libSkiaSharp.so.99.9 /app/lib/libSkiaSharp.so
```
Comment (hfiguiere) [inline io.gitlab.android_translation_layer.BaseApp.yml:81]: so what happen on aarch64?

File: io.gitlab.android_translation_layer.BaseApp.yml [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+  - '/lib/*.a'
+  - '/lib/*.la'
+  - '/lib/libopenxr_loader.so*'
+  - '/lib/libportal-*'
+  - /lib/cmake
+  - /lib/girepository-1.0
+  - /lib/java/core-all_classes.jar
+  - /lib/java/dx.jar
+  - /lib/pkgconfig
+  - /lib/python3.11
```
Comment (hfiguiere) [inline io.gitlab.android_translation_layer.BaseApp.yml:20]: except that this runtime has Python 3.12

File: io.gitlab.android_translation_layer.BaseApp.yml [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+runtime-version: '47'
+sdk: org.gnome.Sdk
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.openjdk17
+cleanup:
+  - /include
+  - /bin/gn
+  - /bin/dx
+  - '/lib/*.a'
+  - '/lib/*.la'
```
Comment (hfiguiere) [inline io.gitlab.android_translation_layer.BaseApp.yml:12]:
```suggestion
  - '*.a'
  - '*.la'
```

File: io.gitlab.android_translation_layer.BaseApp.yml [hunk 5]
```diff
@@ -0,0 +1,261 @@
+id: io.gitlab.android_translation_layer.BaseApp
+runtime: org.gnome.Platform
```
Comment (hfiguiere) [inline io.gitlab.android_translation_layer.BaseApp.yml:2]: is GNOME really necessary?

Comment (julianwi) [reply io.gitlab.android_translation_layer.BaseApp.yml:2]: Yes, we use GTK4. The android View hierarchy will be translated into a GTK4 widget hierarchy. Also we use WebkitGTK for Android apps requiring webkit

File: io.gitlab.android_translation_layer.BaseApp.yml (hunk 1 above)
Comment (hfiguiere) [reply io.gitlab.android_translation_layer.BaseApp.yml:218]: the last runtime that supported it was EOL in 2021. in short it's not supported on flathub

File: io.gitlab.android_translation_layer.BaseApp.yml [hunk 6]
```diff
@@ -0,0 +1,261 @@
...
+  - /lib/java/core-all_classes.jar
+  - /lib/java/dx.jar
+  - /lib/pkgconfig
+  - /lib/python3.11
+  - /share/doc
+  - /share/gir-1.0
+  - /share/man
+  - /share/vala
+modules:
+  - name: gn
```
[the remark this answers is not in the document]
Comment (julianwi) [reply io.gitlab.android_translation_layer.BaseApp.yml:26]: Yes gn is only required as build system for skia. It gets removed in the clean step

[the remark this answers is not in the document]
Comment (hfiguiere) [reply io.gitlab.android_translation_layer.BaseApp.yml:26]: the module should have a `'*'` cleanup rule instead.

**Comment by julianwi** [comment]
> NewPipe issue says:
> 
> > The first launch will take a few seconds to do the ahead-of-time compilation.
> 
> Can this be handled during build time?

Should be possible. This is also something I wanted to investigate in the future

File: io.gitlab.android_translation_layer.BaseApp.yml (hunk 1 above)
Comment (julianwi) [reply io.gitlab.android_translation_layer.BaseApp.yml]: applied

File: io.gitlab.android_translation_layer.BaseApp.yml (hunk 2 above)
Comment (julianwi) [reply io.gitlab.android_translation_layer.BaseApp.yml]: I have fixed it now by getting the correct value from `$FLATPAK_ARCH`

File: io.gitlab.android_translation_layer.BaseApp.yml (hunk 3 above)
Comment (julianwi) [reply io.gitlab.android_translation_layer.BaseApp.yml]: I have fixed it by simply adding cleanup '*' to the python3-typogrify module

File: io.gitlab.android_translation_layer.BaseApp.yml (hunk 4 above)
Comment (julianwi) [reply io.gitlab.android_translation_layer.BaseApp.yml]: applied

File: io.gitlab.android_translation_layer.BaseApp.yml (hunk 6 above)
[the remark this answers is not in the document]
Comment (julianwi) [reply io.gitlab.android_translation_layer.BaseApp.yml:24]: done

File: io.gitlab.android_translation_layer.BaseApp.yml [hunk 7]
```diff
@@ -0,0 +1,261 @@
...
+  - /lib/girepository-1.0
+  - /lib/java/core-all_classes.jar
+  - /lib/java/dx.jar
+  - /lib/pkgconfig
+  - /lib/python3.11
+  - /share/doc
+  - /share/gir-1.0
+  - /share/man
+  - /share/vala
+modules:
```
[the remark this answers is not in the document]
Comment (julianwi) [reply io.gitlab.android_translation_layer.BaseApp.yml:23]: done

File: io.gitlab.android_translation_layer.BaseApp.yml (hunk 6 above)
[the remark this answers is not in the document]
Comment (bbhtt) [reply io.gitlab.android_translation_layer.BaseApp.yml:24]: Is `gn` needed for building apps that are going to use this baseapp ? If so cleaning it here means they can't use it


Or is it a build dependency of things in the baseapp ?

File: io.gitlab.android_translation_layer.BaseApp.yml [hunk 8]
```diff
@@ -0,0 +1,275 @@
...
+    buildsystem: simple
+    build-options:
+      append-path: /usr/lib/sdk/openjdk17/bin
+      env:
+        JAVA_HOME: /usr/lib/sdk/openjdk17/jvm/openjdk-17
+        LIBRARY_PATH: /app/lib
+    build-commands:
+      - make -j8
+      - make ____PREFIX=/app ____LIBDIR=lib ____INSTALL_ETC=/app/etc install
+    sources:
```
Comment (bbhtt) [inline io.gitlab.android_translation_layer.BaseApp.yml:255]: use `no-autogen: true` and `make-install-args` with the options, drop `-j8`, build-commands and `buildsystem: simple`

File: io.gitlab.android_translation_layer.BaseApp.yml (hunk 6 above)
[the remark this answers is not in the document]
Comment (julianwi) [reply io.gitlab.android_translation_layer.BaseApp.yml:24]: It is only used for building skia in the baseapp. I don't know of anything else using it

File: io.gitlab.android_translation_layer.BaseApp.yml (hunk 8 above)
Comment (julianwi) [reply io.gitlab.android_translation_layer.BaseApp.yml]: done

**Comment by bbhtt** [comment]
bot, build io.gitlab.android_translation_layer.BaseApp

**Comment by julianwi** [comment]
bot, build io.gitlab.android_translation_layer.BaseApp

**Comment by julianwi** [comment]
bot, build io.gitlab.android_translation_layer.BaseApp

**Comment by julianwi** [comment]
bot, build io.gitlab.android_translation_layer.BaseApp

**Comment by julianwi** [comment]
bot, build io.gitlab.android_translation_layer.BaseApp

**Comment by julianwi** [comment]
The arm64 patches have been applied to our gitlab repos

**Comment by julianwi** [comment]
bot, build io.gitlab.android_translation_layer.BaseApp

**Pull request closed without merging**